### PR TITLE
Improving the front page README

### PR DESCRIPTION
**Pull request by marstr**
- Removing the weird, out of date, poorly formatted package list.
- Writing a little information about how we do versioning.
- Overhauling the documentation section.
- Moving the installation section to the top.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,128 +1,60 @@
...
 
-# Documentation
+In repositories that are children of this one, [storage for example](https://github.com/Azure/azure-storage-go), we
+have adopted SemVer.org's recommendations.
 
-Read the Godoc of the repository at [Godoc.org](http://godoc.org/github.com/Azure/azure-sdk-for-go/).
+## Azure Versions
+Azure services _mostly_ do not use SemVer based versions. Rather, they use profiles identified by dates. One will often
+see this casually referred to as an "APIVersion". At the moment, our SDK only supports the most recent profiles. In
+order to lock to an API version, one must also lock to an SDK version. However, as discussed in #517, our objective is
```
Comment (mcardosos) [inline README.md]: 517 link missing

File: README.md [hunk 2]
```diff
@@ -1,128 +1,60 @@
...
+for Go uses [glide](https://github.com/Masterminds/glide).
 
-```
-$ cd your/project
-$ glide create
-$ glide install
-```
+# Versioning
+## SDK Versions
+The tags in this repository are based on, but do not conform to [SemVer.org's recommendations](http://semver.org/).
```
Comment (mcardosos) [inline README.md]: I am sure this section can be written in a different way. Thoughts 
@kirthik ?

File: README.md (hunk 1 above)
Comment (marstr) [inline README.md]: Ahh, dang, I thought GitHub would automatically do this.

File: README.md (hunk 2 above)
Comment (marstr) [inline README.md]: It's embarrassing, but it is the truth. I'd rather be slightly self-deprecating than slightly misleading. :)

File: README.md (hunk 1 above)
Comment (marstr) [inline README.md]: Fixed locally on my machine

File: README.md [hunk 3]
```diff
@@ -1,128 +1,60 @@
...
+order to lock to an API version, one must also lock to an SDK version. However, as discussed in #517, our objective is
+to reorganize and publish independent packages for each profile. In that way, we'll be able to have parallel support in
+a single SDK version for all APIVersions supported by Azure.
 
-# Contribute
+# Documentation
 
-If you would like to become an active contributor to this project please follow the instructions provided in [Microsoft Azure Projects Contribution Guidelines](http://azure.github.io/guidelines/).
+- Azure SDK for Go Documentation is available at [GoDoc.org](http://godoc.org/github.com/Azure/azure-sdk-for-go/).
+- Azure REST APIs used by packages in this repository, are documented at [Microsoft Docs, Azure REST](https://docs.microsoft.com/en-us/rest/api/).
```
Comment (marstr) [inline README.md]: There's an extra comma here.

Comment (marstr) [inline README.md]: Fixed locally

**Pull request merged**